projects/nico778/proposal/README.md

```diff
@@ -0,0 +1,41 @@
+Personally the use of carbon fiber and its possibilities has always fascinated me. Particularly well made DIY solutions. Since the cost of a capable 3d printers is going down year after year, the idea of reliable selfmade carbon fiber parts is slowly becoming reality. 
+
+
+## Proposal
+
+A simplistic slicing algorithm will be applied to an already existing 3d mesh. Based on my understanding of geometry processing, a slicing algorithm falls in this field because we are only applying mathematical operations in order to transform a geometric object into multiple geometric objects. The single slices will then be displayed as independent 3d meshes. As this project is implementing a very naive slicing algorithm, this proposal is not related to a research paper. There are solutions that implement such a naive algorithm, such as the referenced github repository.
+I was also looking at various other, much better solutions presented in research papers. However, they far exceed my knowledge of mathematics and geometry processing and I did not even fully the research paper. So, implementing it would probably be impossible. My project will be different from the referenced one because I will be using a different programming language, etc.
+
+
+## Implementation
+
+Unlike in the referenced github repository, this implementation will be implemented with the same tech stack as in this course(VSCode/javascript/node.js/three.js).
+Originally(and ultimately) my plan was using C++, OpenGL and the QT Framework. But because I have the same amount of javascript and C++ experience(very little), javascript seems to be the better route.
+Like already mentioned, the proposed approach is the obvious naive solution. Using a plane as reference for the slicing process, we go through all the faces of the triangle mesh. We can then calculate which faces of the mesh are on which side of the plane. Based on that we the create the two resulting meshes. These meshes are not finished. We have to iterate through all faces of the original mesh that are intersected by the plane.
+These faces are split into 3 new faces(see section Implementation[1]). The main difficulty will probably be the building of these boundary faces. Considering my struggle with HW4, this part will take the most time. 
+Lastly our new meshes, whose number depends on the slice thickness) will be displayed as separate slices.
+Should this project turn out to be too easy(I doubt it), the algorithm could be improved "indefinitely" to handle genus > 0 meshes[3]/more complex meshes, and work faster.
```

Comment:
Intersecting with triangles is a well-understood solver, therefore I can imagine the most complexity is to deal with the internal representation itself.

If you can either:

1. handle any arbitrary drawing stroke (i.e. curve) slicing
2. handle non-manifold triangle meshes (which we partially covered in the course)
3. handle quad meshes (in this way you will share common tasks with other of your classmates, talk to them :)

It will be a challenging but definitely doable and excellent individual project.